main.go

```diff
@@ -53,88 +77,71 @@ func main() {
+	if err != nil {
+		log.Fatal(errors.Wrap(err, "Unable to create the messaging listen client"))
 	}
+	defer listenClient.Close()
 
-	rootResourceNames := config.GetStringSlice("icat.rootResources")
-	if rootResourceNames == nil {
-		log.Fatal("icat.rootResources must be set in the configuration file")
+	publishClient, err := messaging.NewClient(configuration.AMQPURI, true)
+	if err != nil {
+		log.Fatal(errors.Wrap(err, "Unable to create the messaging publish client"))
 	}
+	defer publishClient.Close()
 
-	//refreshInterval, err := time.ParseDuration(*refreshIntervalFlag)
-	_, err = time.ParseDuration(*refreshIntervalFlag)
+	log.Info(configuration.AMQPExchangeName)
+	err = publishClient.SetupPublishing(configuration.AMQPExchangeName)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatal(errors.Wrap(err, "Unable to set up message publishing"))
 	}
 
-	dbconn = sqlx.MustConnect("postgres", dbURI)
-	icatconn = sqlx.MustConnect("postgres", icatURI)
-
-	app := api.New(dbconn, dbSchema, icatconn, userSuffix)
-
-	//workerConfig := worker.Config{
-	//	Name:                    strings.ReplaceAll(uuid.New().String(), "-", ""),
-	//	ExpirationInterval:      workerLifetime,
-	//	RefreshInterval:         refreshInterval,
-	//	WorkerPurgeInterval:     purgeWorkersInterval,
-	//	WorkSeekerPurgeInterval: purgeSeekersInterval,
-	//	WorkClaimPurgeInterval:  purgeClaimsInterval,
-	//	ClaimLifetime:           claimLifetime,
-	//	WorkSeekingLifetime:     seekingLifetime,
-	//	NewUserTotalInterval:    newUserTotalInterval,
-	//}
-
-	//log.Infof("worker name is %s", workerConfig.Name)
-
-	//w, err := worker.New(context.Background(), &workerConfig, dbconn)
-	//if err != nil {
-	//	log.Fatal(err)
-	//}
-
-	//log.Infof("worker ID is %s", w.ID)
-
-	//go w.Start(context.Background())
-
-	//dedb := db.NewDE(dbconn, dbSchema)
-	//usage, err := dedb.AddUserDataUsage(context.Background(), "mian@iplantcollaborative.org", 12345678, time.Now())
-	//if err != nil {
-	//	log.Info(err)
-	//}
-	//log.Info(usage)
-
-	//icattx, err := icatconn.BeginTxx(context.Background(), nil)
-	//if err != nil {
-	//	log.Fatal(err)
-	//}
-	//icatdb := db.NewICAT(icattx, userSuffix, "cyverse")
-	//usage, err := icatdb.UserCurrentDataUsage(context.Background(), "mian", rootResourceNames)
-	//if err != nil {
-	//	log.Error(err)
-	//}
-	//log.Info(usage)
+	go listenClient.Listen()
+
+	queueName := getQueueName(configuration.AMQPQueuePrefix)
+	listenClient.AddConsumerMulti(
+		configuration.AMQPExchangeName,
+		configuration.AMQPExchangeType,
+		queueName,
+		[]string{"index.all", "index.usage.data", "index.usage.data.batch.user.#", a.SingleUserPrefix + ".#"},
+		func(del amqp.Delivery) {
+			var err error
+			log.Tracef("Got message: %s", del.RoutingKey)
+			if del.RoutingKey == "index.all" || del.RoutingKey == "index.usage.data" {
+				// generate prefixes and publish
+			} else if strings.HasPrefix(del.RoutingKey, "index.usage.data.batch.user") {
+				// handle batch
```

Comment:
Is this condition and the condition above going to be handled in a later PR, or maybe I missed some other way of handling these 2 cases?

Reply:
Yeah, these are for later PRs. The first one will fetch all usernames from the DB(s) and publish a bunch of `index.usage.data.batch.user.*` messages. The second one will consume one of those messages and update the data usage for everything within the batch (or possibly just everything whose current info is older than the refresh interval). But, not done yet. For now it'll just acknowledge the messages and carry on.